cmd/kube-apiserver/app/aggregator.go

```diff
@@ -146,17 +146,56 @@ func createAggregatorServer(aggregatorConfig *aggregatorapiserver.Config, delega
 }
 
 func makeAPIService(gv schema.GroupVersion) *apiregistration.APIService {
+	apiServicePriority, ok := apiVersionPriorities[gv]
+	if !ok {
+		// if we aren't found, assume we're a CRD
+		apiServicePriority = priority{group: 5000, version: 20}
+	}
 	return &apiregistration.APIService{
 		ObjectMeta: metav1.ObjectMeta{Name: gv.Version + "." + gv.Group},
 		Spec: apiregistration.APIServiceSpec{
 			Group:           gv.Group,
 			Version:         gv.Version,
-			GroupPriority:   1000,
-			VersionPriority: 10,
+			GroupPriority:   apiServicePriority.group,
+			VersionPriority: apiServicePriority.version,
 		},
 	}
 }
 
+type priority struct {
+	group   int32
+	version int32
+}
+
+// The proper way to resolve this problem is to refactor the genericapiserver.DelegationTarget to include a list of priorities based on which APIs were installed.
+// This requires the APIGroupInfo struct to evolve and include the concept of priorities and to avoid mistakes, the core storage map there needs to be updated.
+// That ripples out every bit as far as you'd expect, so for 1.7 we'll include the list here instead of being built up during storage.
```

Comment:
I'm really unhappy losing strong VersionPreferenceOrder. At least include a breadcrumb in the place we build the storage map until we can refactor it to preserve the info.